subgraphs/vesta-finance/src/entities/stabilitypool.ts

```diff
@@ -1,42 +1,31 @@
-import { Address, BigDecimal, ethereum } from "@graphprotocol/graph-ts";
-import { _StabilityPool } from "../../generated/schema";
-import { updateProtocolUSDLocked } from "./protocol";
-import { BIGDECIMAL_ZERO } from "../utils/constants";
+import { Address, ethereum, BigInt } from "@graphprotocol/graph-ts";
+import { VST_ADDRESS } from "../utils/constants";
+import { bigIntToBigDecimal } from "../utils/numbers";
+import { getOrCreateStabilityPool } from "./market";
+import { getOrCreateAssetToken, getVSTTokenPrice } from "./token";
 
-export function getStabilityPool(asset: Address): _StabilityPool | null {
-  const id = asset.toHexString();
-  return _StabilityPool.load(id);
-}
-
-export function createStabilityPool(asset: Address): _StabilityPool {
-  const id = asset.toHexString();
-  const stabilityPool = new _StabilityPool(id);
-  stabilityPool.id = asset.toHexString();
-  stabilityPool.totalValueLocked = BIGDECIMAL_ZERO;
-  stabilityPool.save();
-  return stabilityPool;
-}
-
-export function getOrCreateStabilityPool(asset: Address): _StabilityPool {
-  const id = asset.toHexString();
-  let stabilityPool = _StabilityPool.load(id);
-  if (stabilityPool == null) {
-    stabilityPool = createStabilityPool(asset);
-  }
-  return stabilityPool;
-}
-
-export function updateStabilityPoolUSDLocked(
+export function updateStabilityPoolTVL(
   event: ethereum.Event,
-  asset: Address,
-  totalValueLocked: BigDecimal
+  VSTAmount: BigInt,
+  assetAmount: BigInt,
+  asset: Address
 ): void {
-  const stabilityPool = getOrCreateStabilityPool(asset);
-  const previousTotalValueLocked = stabilityPool.totalValueLocked;
-  stabilityPool.totalValueLocked = totalValueLocked;
-  updateProtocolUSDLocked(
-    event,
-    totalValueLocked.minus(previousTotalValueLocked)
+  const VSTPrice = getVSTTokenPrice(event);
+  const vstToken = getOrCreateAssetToken(Address.fromString(VST_ADDRESS));
+  vstToken.lastPriceUSD = VSTPrice;
+  vstToken.lastPriceBlockNumber = event.block.number;
+  vstToken.save();
+
+  const VSTValue = bigIntToBigDecimal(VSTAmount).times(VSTPrice);
+  const assetToken = getOrCreateAssetToken(asset);
+  const totalAssetValue = bigIntToBigDecimal(assetAmount).times(
+    assetToken.lastPriceUSD!
   );
+  const stabilityPoolTVL = VSTValue.plus(totalAssetValue);
+  const stabilityPool = getOrCreateStabilityPool(event.address, asset, event);
+  stabilityPool.inputTokenBalance = VSTAmount;
+  stabilityPool.totalValueLockedUSD = stabilityPoolTVL;
+  stabilityPool.totalDepositBalanceUSD = stabilityPoolTVL;
```

Comment:
Should we make different the TVL and depositBalance? Users only deposit VST into the stability pool. The asset amount is given by the protocol to the users in exchange of their deposits.

So, TVL = vst + assetAmount, and deposit = vst?

Reply:
Oh, just saw that for liquity I made it the same way 😅 

What do you think? Should we change it?

Reply:
That's a good point! It makes sense to change it.

Reply:
We can leave it as is then 👍